File: algorithms/1_easy/L/02899_last_visited_integers/solution.py
class Solution:
    def lastVisitedIntegers(self, nums: List[int]) -> List[int]:
        seen = []
        consecutive_negative_ones = 0
        res = []

        for num in nums:
            if num == -1:
                consecutive_negative_ones += 1

                if consecutive_negative_ones > len(seen):
                    res.append(-1)
                else:
                    res.append(seen[len(seen) - consecutive_negative_ones])
            else:
                seen.append(num)
                consecutive_negative_ones = 0

        return res

File: algorithms/1_easy/L/02899_last_visited_integers/solution.cpp
class Solution {
public:
    vector<int> lastVisitedIntegers(vector<int>& nums) {
        vector<int> seen{};
        size_t consecutive_negative_ones{0};
        vector<int> res{};

        for (const int& num : nums) {
            if (num == -1) {
                consecutive_negative_ones += 1;

                if (consecutive_negative_ones > seen.size()) {
                    res.push_back(-1);
                } else {
                    res.push_back(seen[seen.size() - consecutive_negative_ones]);
                }
            } else {
                seen.push_back(num);
                consecutive_negative_ones = 0;
            }
        }

        return res;
    }
};
